google/cloud/firestore_v1/base_collection.py

```diff
@@ -160,9 +161,9 @@ def _parent_info(self) -> Tuple[Any, str]:
     def _prep_add(
         self,
         document_data: dict,
-        document_id: str = None,
-        retry: retries.Retry = None,
-        timeout: float = None,
+        document_id: Optional[str] = None,
```

Comment:
why are all of these changing to optional?

Reply:
it is already "optional" before. By adding the default value (`document_id=None`), it means the `document_id` does not need to be passed when we call `_prep_add`. The `Optional[str]` syntax means: it can either be `str` type or `None` type.

So it was in fact a bug before that the annotation was only `str` and not `Optional[str]`.

Reply:
More accurately, it is not a bug, but a change in the `mypy` type checker itself.
Prior to `mypy` 0.991, implicit optional was allowed. 
```
def foo(s: str = None) -> int: 
```
In the above example, the `s` parameter is optional (can be None), but the type was specified as `str`. This is what's considered as "implicit" optional.

Starting mypy 0.991, by default `mypy` will only accept "explicit" optional type.
```
def foo(s: str = Optional[str] = None): ->:
```
Which explicitly states that `s` parameter must either be a string `str` or `None`.

Reply:
awesome, thanks for the explanation!